packages/marketplace/src/components/ui/developer-webhooks/webhook-edit-modal.tsx

```diff
@@ -0,0 +1,252 @@
+  return topics.map(
+    topic =>
+      ({
+        value: topic.id,
+        label: topic.name,
+        description: topic.description,
+      } as SelectOption),
+  )
+}
+
+export const generateCustomerOptions = (customers: CustomerItem[]) => {
+  const customerOptions: SelectOption[] = [
+    {
+      value: 'SBOX',
+      label: 'SBOX',
+      description: 'SBOX',
+    } as SelectOption,
+  ]
+  customers.forEach((customer: CustomerItem) => {
+    if (customer.status === 'Active') {
+      customerOptions.push({
+        value: customer.client,
+        label: customer.client,
+        description: customer.client,
+      } as SelectOption)
+    }
+  })
+  return customerOptions
+}
+
+export const onCreate = (dispatch: Dispatch, appId: string) => (values: FormValuesType) => {
+  const params: CreateWebhookParams = {
+    applicationId: appId,
+    url: values.url,
+    description: '',
+    topicIds: values.topicIds,
+    customerIds: values.customerIds,
+    active: values.active,
+  }
+  dispatch(createWebhook(params))
+}
+
+export const onEdit = (dispatch: Dispatch, webhookId: string, appId: string) => (values: FormValuesType) => {
+  const params: EditWebhookParams = {
+    applicationId: appId,
+    webhookId,
+    url: values.url,
+    description: '',
+    topicIds: values.topicIds,
+    customerIds: values.customerIds,
+    active: values.active,
+  }
+  dispatch(editWebhook(params))
+}
+
+export const validateURL = (value: string): string | null => {
+  return isValidHttpsUrl(value) ? null : 'The value must be a valid and secure URI'
+}
+
+export const WebhookEditModal: React.FunctionComponent<WebhookEditProps> = ({
+  isUpdate = false,
+  appId,
+  visible,
+  webhookId = '',
+  closeModal,
+  afterClose,
+}) => {
+  const dispatch = useDispatch()
+  const topics = useSelector(selectTopics)
+  const customers = useSelector(selectCustomers)
+  const loading = useSelector(selectLoading)
+  const webhookData = useSelector(selectWebhookData)
+
+  const modalConfig = isUpdate ? EDIT_MODAL : CREATE_MODAL
+
+  const initFormValues: FormValuesType = {
+    url: webhookData?.url,
+    topicIds: webhookData?.topicIds,
+    customerIds: webhookData?.customerIds,
+    active: webhookData?.active,
+  }
+
+  const topicOptions: SelectOption[] = generateTopicOptions(topics)
+  const customerOptions: SelectOption[] = generateCustomerOptions(customers)
+
+  useEffect(() => {
+    if (isUpdate) {
+      dispatch(requestWebhookData(webhookId))
+    } else {
+      dispatch(requestWebhookSubcriptionData(appId))
+    }
+    return () => {
+      dispatch(webhookDataClear())
+    }
+  }, [])
+
+  const onSubmit = isUpdate ? onEdit(dispatch, webhookId, appId) : onCreate(dispatch, appId)
+  const onDelete = () => dispatch(deleteWebhook({ webhookId, applicationId: appId }))
+
+  if (!visible) {
+    return null
+  }
+
+  return (
+    <Modal visible={visible} renderChildren afterClose={afterClose}>
```

Comment:
I think should remove modal if it's not visible
```
visible && <Modal visible={visible}
```

Reply:
We will return null if visible is false (line 156)
```
 if (!visible) {
    return null
  }
```

Reply:
@nphivu414 ok